src/cargo/core/registry.rs

```diff
@@ -166,6 +168,10 @@ impl<'cfg> PackageRegistry<'cfg> {
         self.add_source(source, Kind::Override);
     }
 
+    pub fn add_to_yanked_whitelist(&mut self, iter: impl Iterator<Item = PackageId>) {
+        self.yanked_whitelist.extend(iter)
```

Comment:
I think instead of adding this function here we can just add to the set in `register_lock` below, right? (only add the `id` argument that is)